src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -101,31 +101,47 @@ protected void BuildInternal(ISchema schema, ContentTypeDefinition contentTypeDe
                 }
 
                 // Check if another builder has already added a field for this part.
-                var existingField = graphType.GetField(partName.ToFieldName());
-                if (existingField != null)
+                var partFieldName = partName.ToFieldName();
+                var partFieldType = graphType.GetField(partFieldName);
+
+                if (partFieldType != null)
                 {
-                    // Add content field types.
-                    foreach (var field in part.PartDefinition.Fields)
+                    // Add dynamic content field types to the static part type.
```

Comment:
It is not static, just a global definition.

Reply:
As merged from main, I updated it